### DISCUSSION TITLE
Unexpected shorting behaviour

### DISCUSSION THREAD

**Discussion by Siebe3271** [stefan-jansen/zipline-reloaded#186] [Q&A]
I've created a strategy that I want not to short (exact replication of Andreas Clenow momentum strategy).

Buying looks like: zipline.api.order_target_percent(asset, 0.05)
Selling: zipline.api.order_target(asset, 0)

After some time, the strategy unexpectedly starts opening shorts. I've also 'set_long_only()' in the initialize function but this doesn't help. Anyone else who experience this problem?

**Comment by MBounouar** [comment]
Futures instrument have a date (before or at expiration) after which they automatically close the position. Can you check that it's not the case?

**Reply by Siebe3271** [reply]
Thanks for your answer!

I'm not trading futures, just a regular SPX index. The strange thing is that with the previous Zipline, this exact same code didn't have this shorting problem. 

I've attached my source code in case you want to have a look. Thank you for your time!
[Momentum Backtest.zip](https://github.com/stefan-jansen/zipline-reloaded/files/11298688/Momentum.Backtest.zip)

**Reply by MBounouar** [reply]
I quickly looked at your code (it's not 100% the code from Andreas Clenow). I tested the momentum strategy, using code from [hsm207/trading_evolved](https://github.com/hsm207/trading_evolved/tree/master/notebooks/12_systematic_momentum) (it uses data from quandl) and no shorting happens.  Hard to say what 's the issue, I would recommend debugging to see when shorting happens.
